static/app/views/dashboards/widgets/areaChartWidget/areaChartWidgetVisualization.tsx

```diff
@@ -0,0 +1,216 @@
+import {useRef} from 'react';
+import {useNavigate} from 'react-router-dom';
+import {useTheme} from '@emotion/react';
+import type {
+  TooltipFormatterCallback,
+  TopLevelFormatterParams,
+} from 'echarts/types/dist/shared';
+import type EChartsReactCore from 'echarts-for-react/lib/core';
+
+import BaseChart from 'sentry/components/charts/baseChart';
+import {getFormatter} from 'sentry/components/charts/components/tooltip';
+import AreaSeries from 'sentry/components/charts/series/areaSeries';
+import LineSeries from 'sentry/components/charts/series/lineSeries';
+import {useChartZoom} from 'sentry/components/charts/useChartZoom';
+import {isChartHovered} from 'sentry/components/charts/utils';
+import type {Series} from 'sentry/types/echarts';
+import {defined} from 'sentry/utils';
+import normalizeUrl from 'sentry/utils/url/normalizeUrl';
+import useOrganization from 'sentry/utils/useOrganization';
+import usePageFilters from 'sentry/utils/usePageFilters';
+
+import {useWidgetSyncContext} from '../../contexts/widgetSyncContext';
+import {formatTooltipValue} from '../common/formatTooltipValue';
+import {formatYAxisValue} from '../common/formatYAxisValue';
+import {ReleaseSeries} from '../common/releaseSeries';
+import type {Meta, Release, TimeseriesData} from '../common/types';
+
+export interface AreaChartWidgetVisualizationProps {
+  timeseries: TimeseriesData[];
+  meta?: Meta;
+  releases?: Release[];
+}
+
+export function AreaChartWidgetVisualization(props: AreaChartWidgetVisualizationProps) {
+  const chartRef = useRef<EChartsReactCore | null>(null);
+  const {register: registerWithWidgetSyncContext} = useWidgetSyncContext();
+
+  const pageFilters = usePageFilters();
+  const {start, end, period, utc} = pageFilters.selection.datetime;
+  const {meta} = props;
+
+  const theme = useTheme();
+  const organization = useOrganization();
+  const navigate = useNavigate();
+
+  let releaseSeries: Series | undefined = undefined;
+  if (props.releases) {
+    const onClick = (release: Release) => {
+      navigate(
+        normalizeUrl({
+          pathname: `/organizations/${
+            organization.slug
+          }/releases/${encodeURIComponent(release.version)}/`,
+        })
+      );
+    };
+
+    releaseSeries = ReleaseSeries(theme, props.releases, onClick, utc ?? false);
+  }
+
+  const chartZoomProps = useChartZoom({
+    saveOnZoom: true,
+  });
+
+  // TODO: There's a TypeScript indexing error here. This _could_ in theory be
+  // `undefined`. We need to guard against this in the parent component, and
+  // show an error.
+  const firstSeries = props.timeseries[0];
+
+  // TODO: Raise error if attempting to plot series of different types or units
```

Comment:
Not that I disagree, but line charts could conceivably plot different types so this is where some behaviour would diverge. Is this restriction intended to satisfy the "area charts show parts of a whole" detail? We'll have to consider how this is handled with the widget builder because I don't think this restriction exists there yet.

Reply:
🤔 is it possible in the Widget Builder to query mismatched data? If yes, how does that work, what does the Y Axis do? If I plot, say, duration _and_ error rate